cli/cli.go

```diff
@@ -584,17 +588,25 @@ func (ec *ExecutionContext) readConfig() error {
 	v.SetDefault("api_paths.config", "v1alpha1/config")
 	v.SetDefault("api_paths.pg_dump", "v1alpha1/pg_dump")
 	v.SetDefault("api_paths.version", "v1/version")
-	v.SetDefault("metadata_directory", "")
+	v.SetDefault("metadata_directory", "metadata")
 	v.SetDefault("migrations_directory", "migrations")
 	v.SetDefault("actions.kind", "synchronous")
 	v.SetDefault("actions.handler_webhook_baseurl", "http://localhost:3000")
 	v.SetDefault("actions.codegen.framework", "")
 	v.SetDefault("actions.codegen.output_dir", "")
 	v.SetDefault("actions.codegen.uri", "")
-	v.AddConfigPath(ec.ExecutionDirectory)
-	err := v.ReadInConfig()
-	if err != nil {
-		return errors.Wrap(err, "cannot read config from file/env")
+	if ec.ConfigFile != "" {
+		configBase := filepath.Base(ec.ConfigFile)
+		configName := strings.TrimSuffix(configBase, filepath.Ext(configBase))
+		configPath := filepath.Dir(ec.ConfigFile)
+		v.SetDefault("version", "1")
+		v.SetDefault("metadata_directory", "")
+		v.SetConfigName(configName)
+		v.AddConfigPath(configPath)
+		err := v.ReadInConfig()
```

Comment:
Will it make sense if we use https://godoc.org/github.com/spf13/viper#Viper.SetConfigFile here ?

Reply:
Ah yes, this makes so much more sense. Updated :+1: